rr_computer_vision/include/endline_detection.hpp

```diff
@@ -0,0 +1,42 @@
+/** @file endline_detection.hpp
+ * @author Waleed Ahmed (w29ahmed)
+ * @author Yuchi(Allan) Zhao
+ */
+
+#ifndef ENDLINE_DETECTION_HPP
+#define ENDLINE_DETECTION_HPP
+
+#include <ros/ros.h>
+#include <image_transport/image_transport.h>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <opencv2/opencv.hpp>
+#include <std_srvs/Trigger.h>
+// #include "msg_srv_names.hpp"
+
+class EndlineDetection
+{
+  public :
+    EndlineDetection(ros::NodeHandle);
+    void ImgCb(const sensor_msgs::ImageConstPtr&);
+    static bool compareContourAreas( std::vector<cv::Point> contour1, std::vector<cv::Point> contour2 );
```

Comment:
Genuine question because i dont know. Why is this a static function?

Reply:
Function that you can use without instantiating the object. It's used as a lambda function in the sorting and that only takes static lambda functions